softoff/softoff.cpp

```diff
@@ -285,13 +285,27 @@ int SoftPowerOff::hostSoftOff(sdeventplus::Event& event)
     // TODO: fix mapping to work around OpenBMC ecosystem deficiencies
     pldm_tid_t pldmTID = static_cast<pldm_tid_t>(mctpEID);
 
+    uint8_t effecterState;
+    auto requestHostTransition =
+        pldm::utils::DBusHandler().getDbusProperty<std::string>(
+            "/xyz/openbmc_project/state/host0", "RequestedHostTransition",
+            "xyz.openbmc_project.State.Host");
+    if (requestHostTransition !=
+        "xyz.openbmc_project.State.Host.Transition.Off")
+    {
+        effecterState = PLDM_SW_TERM_GRACEFUL_RESTART_REQUESTED;
+    }
```

Comment:
can we use a ternary operator here?

> effecterState = (requestHostTransition != "xyz.openbmc_project.State.Host.Transition.Off")
>     ? PLDM_SW_TERM_GRACEFUL_RESTART_REQUESTED
>     : PLDM_SW_TERM_GRACEFUL_SHUTDOWN_REQUESTED;

Reply:
why was soft off involved in code update ?